tf2onnx/rewriter/eye_rewriter.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""
+tf2onnx.rewriter.eye_rewriter - supports tf.eye
+"""
+
+from onnx import onnx_pb
+from tf2onnx.graph_matcher import OpTypePattern, GraphMatcher
+
+# pylint: disable=invalid-name,unused-argument,missing-docstring, unused-variable
+
+
+def rewrite_eye(g, ops):
+    # tf.eye is implemented by a sub_graph which contains op "MatrixDiag" or "MatrixSetDiag" while
+    # these two ops are un-supported directly in onnx
+    # but onnx op EyeLike can be used to map the sub_graph
+    # "rewrite_eye" supports tf.eye(tf.shape(x)[i]) and tf.eye(tf.shape(x)[i], tf.shape(x)[j]).
+
+    # ConstantOfShape in opset 9 is used, so if opset less than 9 then do nothing
+    if g.opset < 9:
+        return g.get_nodes()
+
+    pattern1 = \
+        OpTypePattern("MatrixDiag", name="output_eye_matrix", inputs=[
+            OpTypePattern("Fill", inputs=[
+                OpTypePattern("Const"),
+                OpTypePattern("ConcatV2", inputs=[
+                    "*",
+                    "*",
+                    OpTypePattern("Pack", inputs=[
+                        OpTypePattern("Minimum", name="min_node")
+                    ])
+                ])
+            ])
+        ])
+    pattern2 = \
+        OpTypePattern("MatrixSetDiag", name="output_eye_matrix", inputs=[
+            OpTypePattern("Fill"),
+            OpTypePattern("Fill", inputs=[
+                OpTypePattern("Const"),
+                OpTypePattern("ConcatV2", inputs=[
+                    "*",
+                    "*",
+                    OpTypePattern("Pack", inputs=[
+                        OpTypePattern("Minimum", name="min_node")
+                    ])
+                ])
+            ])
+        ])
+
+    for pattern in [pattern1, pattern2]:
+        matcher = GraphMatcher(pattern, allow_reorder=True)
+        match_results = list(matcher.match_ops(ops))
+        for match_result in match_results:
```

Comment:
do we now fully support tf.eye or still partially?
if partially, may need to add some checks for unsupported tf.eye cases and stop rewrite.

Reply:
partially support.

the scenario is not supported when col and row are cost because the specified graph pattern wouldn't match it. and an new op handler will be better than rewriter.

I will support it in next PR.